### PR TITLE
docs: add token and configmaps design

### PR DESCRIPTION
**Pull request by codingben**
Add a new Token and ConfigMap API design.

- `Token` will generate a new token from the service account.
- `ConfigMap` will get data of a specific config map.

### CODE REVIEW

File: docs/api/token.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    {
+      "durationSeconds": int
+    }
+    ```
+  - **Response**: Token from the created service account.
+    ```json
+    {
+       "token": "string",
+    }
+    ```
```
Comment (mzeevi) [inline docs/api/token.md:19]: Shouldn't there be 2 routes here? One that creates a new `serviceAccount` and `token` and one that given a `serviceAccount` `namespacedName` returns its `token`?

**Review by mzeevi — CHANGES_REQUESTED** [review]
Take a look as well @fran2810

File: docs/api/token.md (hunk 1 above)
Comment (codingben) [reply docs/api/token.md:19]: I think that one router is enough. We can check if service account exists, and if yes, create a token from it. If not, create a new service account and get a token from it.

**Comment by codingben** [comment]
As we discussed @mzeevi and @fran2810, we'll create endpoints to:

- Create a new service account (namespace parameter only)
- Get or create a new token from the service account (namespace and service account name parameters)

**Comment by codingben** [comment]
@mzeevi @fran2810 Can you please review it again?

**Review by fran2810 — APPROVED** [review]
/lgtm

**Pull request merged**